### ISSUE TITLE
Option `--ip` does not (seem to) work.

### ISSUE THREAD

**Issue by auzias** [docker/swarm#1828]
On a four swarm agents cluster, a custom overlay network is run with the [weave plugin](http://www.weave.works/products/weave-net/):
````
user@swarm-master:~$swarm-docker network create --driver weave --subnet 192.168.0.0/16 custom-weave
user@swarm-master:~$swarm-docker network inspect custom-weave
[
    {
        "Name": "custom-weave",
        "Id": "653fda2dbb04275604b816f754001ab2bfac21516021ed959c0fd5b7bf0b17eb",
        "Scope": "global",
        "Driver": "weave",
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "192.168.0.0/16"
                }
            ]
        },
        "Containers": {},
        "Options": {}
    }
]
````

When containers are run their IP address should be set using the `--ip` option. According to the `docker-run` doc:

       --ip=""
          Sets the container's interface IPv4 address (e.g. 172.23.0.9)

       It can only be used in conjunction with --net for user-defined networks

Here is the command performed and its output:
````
user@swarm-master:~$ swarm-docker run --rm -it --net=custom-weave --ip=192.168.0.5 debian:jessie bash -c "ip addr show"
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
74: ethwe0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1410 qdisc noqueue state UP group default 
    link/ether ee:c2:9c:6c:c3:6c brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.2/16 scope global ethwe0
       valid_lft forever preferred_lft forever
    inet6 fe80::ecc2:9cff:fe6c:c36c/64 scope link tentative 
       valid_lft forever preferred_lft forever
76: eth1: <NO-CARRIER,BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:ac:12:00:02 brd ff:ff:ff:ff:ff:ff
    inet 172.18.0.2/16 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe12:2/64 scope link tentative 
       valid_lft forever preferred_lft forever
````

The user defined `--net` option is set, as well as `--ip=192.168.0.5`. While the container is expected to have its `ethwe0` interface IP address set to **192.168.0.5** it is actually set to **192.168.0.2**.

Where does this process is wrong?
The network does not seem *user-defined*/*user configured subnets*. How to do so? According to the doc, the `custom-weave` is one.

       --net="bridge"
          Set the Network mode for the container
                                      [ . . ]
                                      '<network-name>|<network-id>': connect to a user-defined network

#### Bug?
`user@swarm-master:~$ swarm-docker network inspect custom-weave` seems unstable and do not return all containers. Performing the command several time in a row actually displays randomly one sole container -- sometimes none.

---
#### Environment details
**`docker version`**
````
Client:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 18:16:19 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 18:16:19 2016
 OS/Arch:      linux/amd64
````
**`docker info`**
````
Containers: 13
 Running: 4
 Paused: 0
 Stopped: 9
Images: 22
Server Version: 1.10.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 64
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file       --net="bridge"
          Set the Network mode for the container
                                      'bridge': create a network stack on the default Docker bridge
                                      'none': no networking
                                      'container:<name|id>': reuse another container's network stack
                                      'host': use the Docker host network stack. Note: the host mode gives the container full access to local system services  such  as  D-bus
       and is therefore considered insecure.
                                      '<network-name>|<network-id>': connect to a user-defined network

Plugins: 
 Volume: local
 Network: weavemesh bridge null host
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 12
Total Memory: 47.25 GiB
Name: bugs10
ID: D2BC:KHHI:XU3R:VZPV:5SYM:HKZE:VCI3:US2A:KXEM:6A7N:V53I:AAZQ
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Cluster store: consul://127.0.0.1:8500
Cluster advertise: 192.168.196.10:2375
````
**`swarm info`**
````
Containers: 34
 Running: 0
 Paused: 0
 Stopped: 0
Images: 23
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 4
 bugs6: 192.168.196.6:2375
  └ Containers: 15
  └ Reserved CPUs: 0 / 12
  └ Reserved Memory: 0 B / 49.62 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.16.0-4-amd64, operatingsystem=Debian GNU/Linux 8 (jessie), storagedriver=aufs
 bugs7: 192.168.196.7:2375
  └ Containers: 3
  └ Reserved CPUs: 0 / 12
  └ Reserved Memory: 0 B / 49.62 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.16.0-4-amd64, operatingsystem=Debian GNU/Linux 8 (jessie), storagedriver=aufs
 bugs8: 192.168.196.8:2375
  └ Containers: 3
  └ Reserved CPUs: 0 / 12
  └ Reserved Memory: 0 B / 49.62 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.16.0-4-amd64, operatingsystem=Debian GNU/Linux 8 (jessie), storagedriver=aufs
 bugs9: 192.168.196.9:2375
  └ Containers: 13
  └ Reserved CPUs: 0 / 12
  └ Reserved Memory: 0 B / 49.62 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.16.0-4-amd64, operatingsystem=Debian GNU/Linux 8 (jessie), storagedriver=aufs
Plugins: 
 Volume: 
 Network: 
CPUs: 48
Total Memory: 198.5 GiB
Name: 22b98c4a84d2
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
WARNING: No cpu shares support
WARNING: No cpuset support
````
**`uname -a`**
````
Linux bugs20 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1+deb8u6 (2015-11-09) x86_64 GNU/Linux
````

**Comment by vieux** [comment]
@auzias can you please do a `docker version` against swarm ? It looks like you are using 1.0.0 and it was fixed in 1.1.0

Thanks

**Comment by auzias** [comment]
@vieux The client is `1.10.0`, while the server is, indeed, `1.0.0`!

###### ---- _An upgrade later_ ----

Works like a charm! Perfect!


Bien vu! Merci :+1:
